File: android_java_mobius/app/src/main/java/com/fjun/android_java_mobius/MainActivityViewBinder.java
package com.fjun.android_java_mobius;

/**
 * View Binder For Main Activity
 */
public interface MainActivityViewBinder {
    void setTemperature(String temperature);

    void showSomethingWentWrong(String error);

    void showWaitingForPosition();

    void showWaitingForWeather();

    void setImageUrl(String imageUrl);
}

File: android_kotlin/app/src/main/java/com/fjun/android_kotlin/MainActivityViewBinder.kt
package com.fjun.android_kotlin

interface MainActivityViewBinder {
    fun setTemperature(temperature: String)
    fun showSomethingWentWrong(error: String)
    fun showWaitingForPosition()
    fun showWaitingForWeather()
    fun setImageUrl(imageUrl: String)
}